hole/poker.go

```diff
@@ -5,7 +5,27 @@ import (
 	"slices"
 )
 
-func cardRune(number, suit int) rune {
+type CardValue int
+
+const (
+	ACE   CardValue = 0
+	TWO   CardValue = 1
+	THREE CardValue = 2
+	FOUR  CardValue = 3
+	FIVE  CardValue = 4
+	SIX   CardValue = 5
+	SEVEN CardValue = 6
+	EIGHT CardValue = 7
+	NINE  CardValue = 8
+	TEN   CardValue = 9
+	JACK  CardValue = 10
+	QUEEN CardValue = 11
+	KING  CardValue = 12
+)
+
+func cardRune[T int | CardValue](num T, suit int) rune {
+	number := int(num)
```

Comment:
Seems a bit weird to be using generics and casting here, can we not just always accept `cardValue` and cast at the call sties? Stuff like `number > 10` should still work with a typed int, right? I'm not sure why we're casting inside the func.

Reply:
the reason is because there are random cases such as are passing ints.

```go
// Pair
for i := 0; i < handCount; i++ {
cards := rand.Perm(13)
suits := rand.Perm(4)
hands = append(hands, Hand{"Pair", []rune{
	cardRune(cards[0], suits[0]),
	...
}})}
```

`cards[0] (variable of type int) as cardValue value in argument to cardRune`
`cannot use ace (constant 0 of type cardValue) as int value in argument to cardRune`

Because no single type signature is valid with the current code, I thought a generic was a good compromise. Which avoids in-line type casts throughout the test cases.